### DISCUSSION TITLE
Solution for using tox on a network shared folder (multiple machines)

### DISCUSSION THREAD

**Discussion by ssbarnea** [tox-dev/tox#1806] [Q&A] [answered]
If you try to run tox from multiple machines that mount the code repository on a shared folder you will fail, mainly because `.tox` folder is not really portable across different machines.

The only workaround I found is to use customize the name of `.tox` folder to make it different on each machine, so they would not overlap. Still, that is only a theoretical approach as you cannot readdy do it.  Even if we would be able to parametrize its name it would end-up cluttering the repository root. We can easily image that having 4-5 extra `.tox*` folders at the root would not really appealing.

Another approach that I was wondering about was a potential use of a system/user temp directory outside the code repository, something like `~/.cache/tox/`. As `~/.cache` is already standardized (XDG) and likely to be not-portable across machines, it could be a good candidate for storing the temporary files. 

In fact this approach is already used by pre-commit tool which keeps its virtualenvs inside ` ~/.cache/pre-commit`, having its own recycling logic. In fac this approach also makes it very easy to cleanup tool temp files for allprojects.

**Comment by gaborbernat** [comment]
In a tox4 world, you can do it as:

```
tox4 --workdir ~/.cache/tox/project
```

**Comment by ssbarnea** [comment, marked as answer]
If I could defined something`TOX_WORKDIR=~/.cache/tox/{projectname}` it would be we awesome. Still, to fully work it needs some kind of expansion as I would never be able to convince 100+ repo-owners to customize their tox.ini files to cover for my corner use-case.

That is the kind of setting that is more of user-setting than a project-setting.

Maybe this could also be implemented as a plugin, as you likely want to avoid bloating the core.

**Reply by gaborbernat** [reply]
> If I could defined something`TOX_WORKDIR=~/.cache/tox/{projectname}` it would be we awesome.

In tox 4 you can do this for all flags. You can also override **all** options within tox.ini via env vars (either via TOX_OVERRIDE env-var or via the -x flag).
